provider/cluster/kube/builder/statefulset.go

```diff
@@ -0,0 +1,80 @@
+package builder
+
+import (
+	"github.com/tendermint/tendermint/libs/log"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	manitypes "github.com/ovrclk/akash/manifest"
+	mtypes "github.com/ovrclk/akash/x/market/types"
+)
+
+type StatefulSet interface {
+	workloadBase
+	Create() (*appsv1.StatefulSet, error)
+	Update(obj *appsv1.StatefulSet) (*appsv1.StatefulSet, error)
+}
+
+type statefulSet struct {
+	workload
+}
+
+var _ StatefulSet = (*statefulSet)(nil)
+
+func NewStatefulSet(log log.Logger, settings Settings, lid mtypes.LeaseID, group *manitypes.Group, service *manitypes.Service) StatefulSet {
+	return &statefulSet{
+		workload: newWorkloadBuilder(log, settings, lid, group, service),
+	}
+}
+
+func (b *statefulSet) Create() (*appsv1.StatefulSet, error) { // nolint:golint,unparam
+	replicas := int32(b.service.Count)
+	falseValue := false
+
+	var effectiveRuntimeClassName *string
+	if len(b.runtimeClassName) != 0 && b.runtimeClassName != runtimeClassNoneValue {
+		effectiveRuntimeClassName = &b.runtimeClassName
+	}
+
+	kdeployment := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   b.Name(),
+			Labels: b.labels(),
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: b.labels(),
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: b.labels(),
+				},
+				Spec: corev1.PodSpec{
+					RuntimeClassName: effectiveRuntimeClassName,
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsNonRoot: &falseValue,
+					},
+					AutomountServiceAccountToken: &falseValue,
+					Containers:                   []corev1.Container{b.container()},
+				},
+			},
+			VolumeClaimTemplates: b.persistentVolumeClaims(),
```

Comment:
Shouldn't the `persistentVolumeClaims()` function be defined on this type since it is only used here?